dipy/align/streamwarp.py

```diff
@@ -278,8 +278,13 @@ def bundlewarp_shape_analysis(moving_aligned, deformed_bundle, no_disks=10,
 
     for i in range(n):
 
-        shape_profile[i] = np.mean(offsets[indx == i])
-        stdv[i] = np.std(offsets[indx == i])
+        mask = indx == i
+        if sum(mask):
```

Comment:
I am not sure if all values being `False` is something that can be expected. It looks weird to me.

Reply:
@BramshQamar, can you answer @jhlegarreta  concerning this point.

Reply:
Hi @jhlegarreta, good catch!

You are right, and it shouldn't return all `False`. I see there is a logical problem in the test function. I set the number of points to 20 in each streamlines https://github.com/dipy/dipy/blob/d2a05444f318cbe1397d492fc83bbd3b898da979/dipy/align/tests/test_streamwarp.py#L62  
and then in the second test case, I am setting n=100, which means 100 segments to be created along the length of the bundle. Since streamlines only have 20 points in them, some segments will be empty (i.e. actual data points are less than the segments).
https://github.com/dipy/dipy/blob/d2a05444f318cbe1397d492fc83bbd3b898da979/dipy/align/tests/test_streamwarp.py#L77

We could make changes in the `def bundlewarp_shape_analysis` function to handle this case and can have another test for it.

I suppose keeping the mask condition should be fine even if the number of points in a streamline is greater or equal to segment/disk numbers.

Reply:
> We could make changes in the def bundlewarp_shape_analysis function to handle this case and can have another test for it.
> I suppose keeping the mask condition should be fine even if the number of points in a streamline is greater or equal to segment/disk numbers.

I am not sure if I see this: if streamlines have N vertices, but you request an analysis over 100, then streamlines should be resampled within this function, regardless of N. Not sure if following this rationale has undesired side-effects.

Reply:
@jhlegarreta We ideally would like to keep the original number of points in a streamline. And we allow multiple points from the same streamline to belong to one segment. We don't expect one point per segment from each streamline. We also allow for a streamline not to have any points belonging to some segments (streamlines of different lengths are not sampled to have the same N number of points). Since this is a test function, I set them to have 20, and passing 100 segment numbers caused an issue here.

Reply:
So then your proposal is to set them to 20 in the test? How do you propose to handle the situation of passing 100: should the method raise an exception or do you have another proposal?